---
title: "s01_explore_icr1000_hc_lof_individuals"
output: html_document
editor_options: 
  chunk_output_type: console
---

Started: Alexey Larionov 16 May 2018  
Updated: David Gilfoyle 18 May 2018

---

# Summary

The ultimate goal of this analysis is to identify individuals with "highest burden" of protein-affecting (LoF) DDR variants.  

A simple sum of multiple LoF DDR variants present in the same person may cause higher impact.  

Homozygous variants may have stronger impact than heterozygous, despite having similar "consequence"/"impact" (all the variants are already HC LoF-s at this stage).  

The assessment of hemizygous or compound heterozygous variants is not possible in the ICR1000 dataset since phasing data is not available

Arguably, a double-LoF-hit in the same allele may be more detremental than a single LoF in the allele. However, the opposite statement may also be considered: that a single HC-LoF already renders enough damage, which can not be further aggravated by additional LoFs).    

Finally, some of the varaints already have Clin-Sig annotations, which will not be considered in detail in this script. This has been assessed during review of VEP annotation for all LoF variants.  

# start_section

```{r start_section}

Sys.time()
rm(list=ls())
graphics.off()

# Install libraries
library(dplyr)
library(knitr)

# Make output wider (default was 80 columns?)
# options(width = 400)

```

# load_rdata_file

```{r load_rdata_file}

# Setup input folder
input_folder="/home/ec2-user/results/filtered/lof_filt/icr1000"

# Load RData file - contains dataframe containing all variants and genotype matrix
load(paste(input_folder, "r01_icr1000_ddr_lof_filtered.RData", sep="/"))

# Keep only HC dataset
rm(fixed_lof_filt_icr1000_ddr.df, fixed_lof_hc_pli_filt_icr1000_ddr.df, gt_add_lof_filt_icr1000_ddr.mx, gt_add_lof_hc_pli_filt_icr1000_ddr.mx, input_folder, output_folder, base_folder)

# List remaining objects
ls()

# Establish rownames in fixed dataframe
fixed_lof_hc_filt_icr1000_ddr.df <- tibble::column_to_rownames(fixed_lof_hc_filt_icr1000_ddr.df, var = "rowname")

# Check remaining dataframe and matrix
dim(fixed_lof_hc_filt_icr1000_ddr.df)
class(fixed_lof_hc_filt_icr1000_ddr.df)
fixed_lof_hc_filt_icr1000_ddr.df[1:20,1:10]

dim(gt_add_lof_hc_filt_icr1000_ddr.mx)
class(gt_add_lof_hc_filt_icr1000_ddr.mx)
gt_add_lof_hc_filt_icr1000_ddr.mx[1:20,1:5]

# Check rowmnames are same 
sum(rownames(gt_add_lof_hc_filt_icr1000_ddr.mx) != rownames(fixed_lof_hc_filt_icr1000_ddr.df)) # should be 0

# Setup base and output folders
base_folder="/home/ec2-user/scripts/data_analysis/gt_analysis"
output_folder="/home/ec2-user/results/data_analysis/gt_analysis"

getwd()
setwd(base_folder)
getwd()

opts_knit$set(root.dir = base_folder)

```

# explore_add_gt_data

```{r explore_add_gt_data}

# explore additive genotypes table
summary(as.factor(gt_add_lof_hc_filt_icr1000_ddr.mx)) # there are no homozygous (2) genotypes

# there are 86 missing genotypes
sum(is.na(gt_add_lof_hc_filt_icr1000_ddr.mx))

```

# explore_loftee_nas

During LOFTEE filtering, it was noted that select variants were 'NA' in the LoF VEP annotation field - these were mainly associated with start_lost and stop_lost variants. Consequently it was considered appropriate to retain these variants in the high confidence LoF filtered dataset.    

```{r explore_loftee_nas}

# Check LoF annotation generated by LOFTEE
summary(as.factor(fixed_lof_hc_filt_icr1000_ddr.df$LoF))

# Look at details of the LOFTEE "NA"s
lof_nas <- which(is.na(fixed_lof_hc_filt_icr1000_ddr.df$LoF))
fixed_lof_hc_filt_icr1000_ddr.df[lof_nas,c("Location", "Consequence", "IMPACT", "LoF")]

# clean-up
rm(lof_nas)

```

# explore_clin_sig

Some of the variants have known clinical significance  

```{r explore_clin_sig}

fixed_lof_hc_filt_icr1000_ddr.df[1:5,c("Location","CLIN_SIG")]
clin_sig <- fixed_lof_hc_filt_icr1000_ddr.df$CLIN_SIG
table(unlist(strsplit(clin_sig, "&")))
rm(clin_sig)

```

# count_variants_burden_per_individual  

Make and keep the named vector for later use  

```{r simple_variants_burden}

# count
simple_burden <- apply(gt_add_lof_hc_filt_icr1000_ddr.mx, 2, sum, na.rm = TRUE)
names(simple_burden) <- colnames(gt_add_lof_hc_filt_icr1000_ddr.mx)

# explore
simple_burden[1:5]
sum(is.na(simple_burden)) # sanity check
table(simple_burden)
which(simple_burden == 2) # shows name and index

```

# function_to_explore_selected_individuals

what genes are affected in selected individuals ?  

```{r explore_selected_individuals}

get_case_summary.udf <- function(case){
  
  # for testing
  #case <- "NA19043"
  
  variants <- which(gt_add_lof_hc_filt_icr1000_ddr.mx[,case] != "0|0" & 
                  gt_add_lof_hc_filt_icr1000_ddr.mx[,case] != "0")
  #variants

  genotypes <- gt_add_lof_hc_filt_icr1000_ddr.mx[variants,case]
  #genotypes

  summary.df <- cbind(fixed_lof_hc_filt_icr1000_ddr.df[variants,c("SYMBOL", "CHROM", "POS", "REF", "ALT", "Consequence")],
                      genotypes,
                      clinsig=fixed_lof_hc_filt_icr1000_ddr.df[variants,c("CLIN_SIG")])
  summary.df

}

```

# explore_selected_individuals

look at what genes are affected in selected individuals

### EGAR00001237641_1522_A06

```{r EGAR00001237641_1522_A06}

get_case_summary.udf("EGAR00001237641_1522_A06")

```

### EGAR00001237663_1522_E04

```{r EGAR00001237663_1522_E04}

get_case_summary.udf("EGAR00001237663_1522_E04")

```

### EGAR00001237778_1524_H09

```{r EGAR00001237778_1524_H09}

get_case_summary.udf("EGAR00001237778_1524_H09")

```

### EGAR00001237792_1525_C02

```{r EGAR00001237792_1525_C02}

get_case_summary.udf("EGAR00001237792_1525_C02")

```

### EGAR00001237957_1528_F10

```{r EGAR00001237957_1528_F10}

get_case_summary.udf("EGAR00001237957_1528_F10")

```

### EGAR00001238096_1620_E06

```{r EGAR00001238096_1620_E06}

get_case_summary.udf("EGAR00001238096_1620_E06")

```

### EGAR00001238110_1620_H02

```{r EGAR00001238110_1620_H02}

get_case_summary.udf("EGAR00001238110_1620_H02")

```

### EGAR00001238190_1622_F05

```{r EGAR00001238190_1622_F05}

get_case_summary.udf("EGAR00001238190_1622_F05")

```

### EGAR00001238285_1626_F08

```{r EGAR00001238285_1626_F08}

get_case_summary.udf("EGAR00001238285_1626_F08")

```

### EGAR00001238358_1628_B09

```{r EGAR00001238358_1628_B09}

get_case_summary.udf("EGAR00001238358_1628_B09")

```

### EGAR00001238383_1628_F10

```{r EGAR00001238383_1628_F10}

get_case_summary.udf("EGAR00001238383_1628_F10")

```

### EGAR00001238397_1628_H12

```{r EGAR00001238397_1628_H12}

get_case_summary.udf("EGAR00001238397_1628_H12")

```

# final_section

```{r final_section}

ls()
sessionInfo()
Sys.time()

```
