---
title: "s01_explore_kgen_hc_lof_individuals"
output: html_document
editor_options: 
  chunk_output_type: console
---

Started: Alexey Larionov 16 May 2018  
Updated: David Gilfoyle 18 May 2018

---

# Summary

The ultimate goal of this analysis is to identify individuals with "highest burden" of protein-affecting (LoF) DDR variants.  

A simple sum of multiple LoF DDR variants present in the same person may cause higher impact.  

Homozygous variants may have stronger impact than heterozygous, despite having similar "consequence"/"impact" (all the variants are already HC LoF-s at this stage).  

X-chromosome hemizygous variants (variants on X chromosome in males) may be as functionaly important as autosomal nomozygous. No genes of interest are located on the Y-chromosome.  

Compound hetrozygous variants could be detected using 1000 Genomes Project phased data. They may also have a higher functional role than simple heterozygous mutations.   

Arguably, just a double-LoF-hit in the same allele may be more detremental than a single LoF in the allele. However, the opposite statement may also be considered: that a single HC-LoF already renders enough damage, which can not be further aggravated by additional LoFs).    

Finally, some of the varaints already have Clin-Sig annotations, which will not be considered in detail in this script. This has been assessed during review of VEP annotation for all LoF variants.  

# start_section

```{r start_section}

Sys.time()
rm(list=ls())
graphics.off()

# Install libraries
library(dplyr)
library(knitr)

# Make output wider (default was 80 columns?)
options(width = 400)

```

# load_rdata_file

```{r load_rdata_file}

# Setup input folder
input_folder="/home/ec2-user/results/filtered/lof_filt/kgen"

# Load RData file - contains dataframe containing all variants and genotype matrix
load(paste(input_folder, "r01_kgen_ddr_lof_filtered.RData", sep="/"))

# Keep only HC dataset
rm(fixed_lof_filt_kgen_ddr.df, fixed_lof_hc_pli_filt_kgen_ddr.df, gt_lof_filt_kgen_ddr.mx, gt_lof_hc_pli_filt_kgen_ddr.mx, input_folder, output_folder, base_folder)

# List remaining objects
ls()

# Establish rownames in fixed dataframe
fixed_lof_hc_filt_kgen_ddr.df <- tibble::column_to_rownames(fixed_lof_hc_filt_kgen_ddr.df, var = "rowname")

# Check remaining dataframe and matrix
dim(fixed_lof_hc_filt_kgen_ddr.df)
class(fixed_lof_hc_filt_kgen_ddr.df)
fixed_lof_hc_filt_kgen_ddr.df[1:20,1:10]

dim(gt_lof_hc_filt_kgen_ddr.mx)
class(gt_lof_hc_filt_kgen_ddr.mx)
gt_lof_hc_filt_kgen_ddr.mx[1:20,1:5]

# Check rowmnames are same 
sum(rownames(gt_lof_hc_filt_kgen_ddr.mx) != rownames(fixed_lof_hc_filt_kgen_ddr.df)) # should be 0

# Setup base and output folders
base_folder="/home/ec2-user/scripts/data_analysis/gt_analysis"
output_folder="/home/ec2-user/results/data_analysis/gt_analysis"

getwd()
setwd(base_folder)
getwd()

opts_knit$set(root.dir = base_folder)

```

# explore_gt_data

```{r explore_gt_data}

# explore genotypes table
summary(as.factor(gt_lof_hc_filt_kgen_ddr.mx)) # there are some aparently gaploid ("0" and "1") genotypes

# explore gaploid genotypes
which(gt_lof_hc_filt_kgen_ddr.mx[,1]=="0") 
fixed_lof_hc_filt_kgen_ddr.df[c(208:210),1:5] # gaploid genotypes are in entries 208 - 210 on X chromosome

# X chromosome variants are gaploid in males and diploid in females 
gt_lof_hc_filt_kgen_ddr.mx[c(208:210),1:5]

# there are no missing genotypes
sum(is.na(gt_lof_hc_filt_kgen_ddr.mx))

```

# explore_loftee_nas

During LOFTEE filtering, it was noted that select variants were 'NA' in the LoF VEP annotation field - these were mainly associated with start_lost and stop_lost variants. Consequently it was considered appropriate to retain these variants in the high confidence LoF filtered dataset.    

```{r explore_loftee_nas}

# Check LoF annotation generated by LOFTEE
summary(as.factor(fixed_lof_hc_filt_kgen_ddr.df$LoF))

# Look at details of the LOFTEE "NA"s
lof_nas <- which(is.na(fixed_lof_hc_filt_kgen_ddr.df$LoF))
fixed_lof_hc_filt_kgen_ddr.df[lof_nas,c("Location", "Consequence", "IMPACT", "LoF")]

# clean-up
rm(lof_nas)

```

# explore_clin_sig

Some of the variants have known clinical significance  

```{r explore_clin_sig}

fixed_lof_hc_filt_kgen_ddr.df[1:5,c("Location","CLIN_SIG")]
clin_sig <- fixed_lof_hc_filt_kgen_ddr.df$CLIN_SIG
table(unlist(strsplit(clin_sig, "&")))
rm(clin_sig)

```

# recode_genotypes_to_additive_numeric

0,1,2 for REF-REF, REF-ALT and ALT-ALT respectively  
Loosing phasing and hemizygous information  

```{r recode_genotypes_to_additive_numeric}

# Prepare matrix of NA-s
gt_add.mx <- matrix(NA,
                    nrow=nrow(gt_lof_hc_filt_kgen_ddr.mx),
                    ncol=ncol(gt_lof_hc_filt_kgen_ddr.mx))

colnames(gt_add.mx) <- colnames(gt_lof_hc_filt_kgen_ddr.mx)
rownames(gt_add.mx) <- rownames(gt_lof_hc_filt_kgen_ddr.mx)

dim(gt_add.mx)
gt_add.mx[1:5,1:5]

# explore the source matrix
gt_lof_hc_filt_kgen_ddr.mx[1:5,1:5]
summary(as.factor(gt_lof_hc_filt_kgen_ddr.mx))

# populate the new matrix
0 -> gt_add.mx[ gt_lof_hc_filt_kgen_ddr.mx == "0"]
0 -> gt_add.mx[ gt_lof_hc_filt_kgen_ddr.mx == "0|0"]
1 -> gt_add.mx[ gt_lof_hc_filt_kgen_ddr.mx == "1"]
1 -> gt_add.mx[ gt_lof_hc_filt_kgen_ddr.mx == "0|1"]
1 -> gt_add.mx[ gt_lof_hc_filt_kgen_ddr.mx == "1|0"]
2 -> gt_add.mx[ gt_lof_hc_filt_kgen_ddr.mx == "1|1"]

# check result
dim(gt_add.mx)
gt_add.mx[1:5,1:5]
summary(as.factor(gt_add.mx))
sum(is.na(gt_add.mx))

```

# count_variants_burden_per_individual  

Make and keep the named vector for later use  

```{r simple_variants_burden}

# count
simple_burden <- apply(gt_add.mx, 2, sum)
names(simple_burden) <- colnames(gt_add.mx)

# explore
simple_burden[1:5]
sum(is.na(simple_burden)) # sanity check
table(simple_burden)
which(simple_burden == 3) # shows name and index

```

# recode_genotypes_to_homo_hemi_zygous

1 for homo- or hemi- zygous  
0 for everything else  

```{r make_homo_hemi}

# prepare matrix of NA-s
gt_homo_hemi.mx <- matrix(NA,
                          nrow=nrow(gt_lof_hc_filt_kgen_ddr.mx),
                          ncol=ncol(gt_lof_hc_filt_kgen_ddr.mx))

colnames(gt_homo_hemi.mx) <- colnames(gt_lof_hc_filt_kgen_ddr.mx)
rownames(gt_homo_hemi.mx) <- rownames(gt_lof_hc_filt_kgen_ddr.mx)

dim(gt_homo_hemi.mx)
gt_homo_hemi.mx[1:5,1:5]

# explore the source matrix
gt_lof_hc_filt_kgen_ddr.mx[1:5,1:5]
summary(as.factor(gt_lof_hc_filt_kgen_ddr.mx))

# populate the new matrix: 
0 -> gt_homo_hemi.mx[ gt_lof_hc_filt_kgen_ddr.mx == "0"]
0 -> gt_homo_hemi.mx[ gt_lof_hc_filt_kgen_ddr.mx == "0|0"]
1 -> gt_homo_hemi.mx[ gt_lof_hc_filt_kgen_ddr.mx == "1"]
0 -> gt_homo_hemi.mx[ gt_lof_hc_filt_kgen_ddr.mx == "0|1"]
0 -> gt_homo_hemi.mx[ gt_lof_hc_filt_kgen_ddr.mx == "1|0"]
1 -> gt_homo_hemi.mx[ gt_lof_hc_filt_kgen_ddr.mx == "1|1"]

# check result
dim(gt_homo_hemi.mx)
gt_homo_hemi.mx[1:5,1:5]
summary(as.factor(gt_homo_hemi.mx))
sum(is.na(gt_homo_hemi.mx))

```

# count_homo_hemi_zygous_variants_per_individual

Make and keep the named vector for later use  

```{r count_homo_hemi}

# count
homo_hemi <- apply(gt_homo_hemi.mx, 2, sum)
names(homo_hemi) <- colnames(gt_homo_hemi.mx)

# explore
homo_hemi[1:5]
sum(is.na(homo_hemi)) # sanity check
table(homo_hemi)
which(homo_hemi == 1) # shows name and index

```

# make_matrix_of_compound_hets 

Aggregate per gene  
1 for compound_het  
0 for everything else  

This chunk uses looping through rows and columns,  
which is considered slow and bad practice in R programming.  
I use loops here for simlicity and because the dataset is small.  

```{r make_compound_hets}

# make empty matrix for results
comp_het_chr.mx <- matrix(nrow=0, ncol=ncol(gt_lof_hc_filt_kgen_ddr.mx)+1)
colnames(comp_het_chr.mx) <- c("gene", colnames(gt_lof_hc_filt_kgen_ddr.mx))

# get list of genes
genes <- unique(fixed_lof_hc_filt_kgen_ddr.df$SYMBOL)
genes[1:5]
length(genes) # genes that dont have LoF variants at all in any dataset may be discussed ...

# Make a compound het in POLQ in HG00096 (for testing)
#"0|1" -> gt_lof_hc_filt_kgen_ddr.mx[36,1]
#"1|0" -> gt_lof_hc_filt_kgen_ddr.mx[37,1]

# for each gene
for(gene in genes){
  
  # select genes for testing
  #gene <- "POLQ" # multiple variants gene
  #gene <- "UBE2T" # single variant gene
  
  # select data for the gene
  gene_rows <- fixed_lof_hc_filt_kgen_ddr.df$SYMBOL == gene
  gene.mx <- gt_lof_hc_filt_kgen_ddr.mx[gene_rows,,drop=F] # note drop=F
  
  # make a compund het (for testing)
  #"0|1" -> gene.mx[1,1]
  #"1|0" -> gene.mx[2,1]
  
  # make vector of for result (assume no compound heterozygous)
  comp_het <- rep(0, ncol(gt_lof_hc_filt_kgen_ddr.mx))
  
  # for genes with more than one variant
  # (no compound heterozygous could be in a single-variant gene)
  if(nrow(gene.mx) > 1){
  
    # for each individual
    for(indiv in 1:ncol(gene.mx)){
      
      # for testing
      #indiv <- 1
      
      # get vector of genotypes for variants in the current gene
      variants <- gene.mx[,indiv]
      
      # if there are variants in both alleles
      if("0|1" %in% variants & "1|0" %in% variants){
        
        # Update the gene's comp_het vector for this individual
        1 -> comp_het[indiv]
        
      } 
    } # next inividual
  }
  
  # compile full result string 
  result <- c(gene, comp_het)
    
  # add result string to result matrix
  comp_het_chr.mx <- rbind(comp_het_chr.mx, result)

} # next gene

# explore and transform comp_het character matrix
comp_het_chr.mx[16:21,1:5]
rownames(comp_het_chr.mx) <- comp_het_chr.mx[,1]
comp_het_chr.mx <- comp_het_chr.mx[,-1]
comp_het_chr.mx[1:5,1:5]
table(comp_het_chr.mx) # no LoF compound hets in kgen  

# convert character matrix to numeric (to keep data type consistent with content)
comp_het.mx <- matrix(as.numeric(comp_het_chr.mx), nrow=nrow(comp_het_chr.mx))
colnames(comp_het.mx) <- colnames(comp_het_chr.mx)
rownames(comp_het.mx) <- rownames(comp_het_chr.mx)

# explore the numeric matrix
comp_het.mx[1:5,1:5]
sum(comp_het.mx)

# clean-up
rm(genes, gene, gene_rows, gene.mx, comp_het, indiv, variants, result, comp_het_chr.mx)

```

# count_compound_hets_per_individual

No compund hets in kgen HC LoF-s  
However, I added this fagment for consistency (may be needed for other subsets?)  

```{r count_compound_hets}

# count
comp_het <- apply(comp_het.mx, 2, sum)
names(comp_het) <- colnames(comp_het.mx)

# explore
comp_het[1:5]
sum(is.na(comp_het)) # sanity check
table(comp_het)
which(comp_het == 1) # shows name and index

```

# merge_results

Merge simple burden, homo-hemi and compound hets into a single table  

```{r}

# make numeric summary matrix
summary.mx <- cbind(simple_burden, homo_hemi, comp_het)
summary.mx[1:5,]
sum(is.na(summary.mx)) # sanity check

# add aggregate score
aggregate_score <- apply(summary.mx, 1, sum)
summary.mx <- cbind(summary.mx, aggregate_score)
summary.mx[1:5,]

# convert numeric matrix to data frame
summary.df <- as.data.frame(summary.mx)
summary.df <- cbind(individual=rownames(summary.mx),summary.df)
summary.df$individual <- as.vector(summary.df$individual)
rownames(summary.df) <- summary.df$individual
str(summary.df)

# show individuals carrying multiple HC LoF variants
summary.df %>% 
  filter(aggregate_score > 1) %>% 
  arrange(desc(aggregate_score), desc(homo_hemi))

# clean-up
rm(simple_burden, homo_hemi, comp_het, aggregate_score, summary.mx)

```

# function_to_explore_selected_individuals

what genes are affected in selected individuals ?  

```{r explore_selected_individuals}

get_case_summary.udf <- function(case){
  
  # for testing
  #case <- "NA19043"
  
  variants <- which(gt_lof_hc_filt_kgen_ddr.mx[,case] != "0|0" & 
                  gt_lof_hc_filt_kgen_ddr.mx[,case] != "0")
  #variants

  genotypes <- gt_lof_hc_filt_kgen_ddr.mx[variants,case]
  #genotypes

  summary.df <- cbind(fixed_lof_hc_filt_kgen_ddr.df[variants,c("SYMBOL", "CHROM", "POS", "REF", "ALT", "Consequence")],
                      genotypes,
                      clinsig=fixed_lof_hc_filt_kgen_ddr.df[variants,c("CLIN_SIG")])
  summary.df

}

```

# explore_selected_individuals

look at what genes are affected in selected individuals

### NA19043

```{r NA19043}

get_case_summary.udf("NA19043")

```

### NA19131

```{r NA19131}

get_case_summary.udf("NA19131")

```

### HG01198

```{r HG01198}

get_case_summary.udf("HG01198")

```

### HG02122

```{r HG02122}

get_case_summary.udf("HG02122")

```

### HG02977

```{r HG02977}

get_case_summary.udf("HG02977")

```

### HG03120

```{r HG03120}

get_case_summary.udf("HG03120")

```

### HG00310

```{r HG00310}

get_case_summary.udf("HG00310")

```

### HG00365

```{r HG00365}

get_case_summary.udf("HG00365")

```

### HG02623

```{r HG02623}

get_case_summary.udf("HG02623")

```

### HG03224

```{r HG03224}

get_case_summary.udf("HG03224")

```

### NA19834

```{r NA19834}

get_case_summary.udf("NA19834")

```

# final_section

```{r final_section}

ls()
sessionInfo()
Sys.time()

```
