---
title: "r01_fixrownames_kgen_ddr"
output: html_document
editor_options: 
  chunk_output_type: console
---

Started: David Gilfoyle 2 May 2018  
Updated: David Gilfoyle 16 May 2018

---

# Summary

Syncronise row names between fixed column and genotype data sets
Dataset: 1000 Genome Project  
Gene Selection: DNA Repair Genes   
Input Variants: 368,639  

# start_section

```{r start_section}

Sys.time()
rm(list=ls())
graphics.off()

# Make sure all the libraries installed 
#install.packages("vcfR")

# Install libraries
library(dplyr)
library(knitr)

# Setup input folder
input_folder="/home/ec2-user/results/r_import/kgen"

```

# load_rdata_file

```{r load_rdata_file}

# Load RData file - contains dataframe containing all variants
load(paste(input_folder, "r01_import_vcf_kgen_ddr.RData", sep="/"))

# Check input fixed column dataframe
dim(fixed_kgen_ddr.df)
class(fixed_kgen_ddr.df)
fixed_kgen_ddr.df[1:20,1:10]

# Check input genotype matrix
dim(gt_kgen_ddr.mx)
class(gt_kgen_ddr.mx)
gt_kgen_ddr.mx[1:5,1:5]

# Setup base and output folders
base_folder="/home/ec2-user/scripts/filtration/qual_filt/kgen"
output_folder="/home/ec2-user/results/filtered/qual_filt/kgen"

getwd()
setwd(base_folder)
getwd()

opts_knit$set(root.dir = base_folder)

```

# syncronise_row_names
During vcfR seperation of genotypes rownames are entered as chromosome/position/row number,  
unless a dbSNP reference is present, in which case row label is dbSNP ref/row number.  
This section aims to enter comparable rownames to the fixed column dataframe to allow selection of  
genotypes following filtering of fixed column dataframe.  

```{r syncronise_row_names}

# Vector of rs-IDs that will be used for "known" variants (with known rs-IDs)
old_vars <- as.vector(fixed_kgen_ddr.df$ID)
old_vars[1:10]

# Vector of labels that will be used for "new" variants (w/o rs-IDs)
new_vars <- paste(fixed_kgen_ddr.df$CHROM, fixed_kgen_ddr.df$POS, sep="_")
new_vars[1:10]

# Write known rs-IDs to the labels
labels <- old_vars
labels[1:10]

# Write "new_vars" to the labels
new_vars[is.na(old_vars)] -> labels[is.na(old_vars)]
labels[1:10]

# Add row numbers to the end of labels
labels <- paste(labels, 1:nrow(fixed_kgen_ddr.df), sep="_")
labels[1:10]

# Assign the rownames to data frame
labels -> rownames(fixed_kgen_ddr.df)

# Check rowmnames are same 
fixed_kgen_ddr.df[1:5,1:10]
gt_kgen_ddr.mx[1:5,1:5]
sum(rownames(gt_kgen_ddr.mx) != rownames(fixed_kgen_ddr.df)) # should be 0
dim(fixed_kgen_ddr.df)

# Clean up
rm(labels, new_vars, old_vars)

```

# save_results

```{r save_results}

save.image(paste(output_folder, "r01_kgen_ddr_updated_rownames.RData", sep="/"))

```

# final_section

```{r final_section}

ls()
sessionInfo()
Sys.time()

```
