---
title: "calculate_CCDS_length"
output: html_document
editor_options: 
  chunk_output_type: console
---

started: Alexey Larionov, 01May2018  
last updated: David Gilfoyle, 11June2018  

# Summary  

### Purpose  

Calculate sum of CDS lengths for DDR and OR genes (to normalise the variant counts)  
**DDR**: 424,018  
**OR**: 323,585  

### Main steps  

- Read CCDS-15 table  
- Read lists of DDR and OR genes  
- Extract DDR and OR genes from CCDS-15 table  
- Calculate CDS length (using a user-defined function)  
- Select longest CDS per gene (consistent with definition of canonical transcript ( https://www.ensembl.org/Help/Glossary?id=346 )
- Calculate sum of the longest CDS for all genes  

### CCDS 15  

CCDS v15 is CCDS for the latest version of GRCH37 (b37.p13).  
"Current" and "original" version are the same for DDR genes (I have not checked for OR).  
Defining CDS is yet an active area of research (see accompanying references).  

### Other relevant resourses in R  

Potentially, we could use already available R packages with b37 genomic gata, such as    
**TxDb.Hsapiens.UCSC.hg19.knownGene** , which contains cds sequencies, along with many other info,   
**org.Hs.eg.db** , which translates entrez, genes symbols, uniprot ids ets or even  
**BSgenome.Hsapiens.UCSC.hg19** , which contains the entire b37 reference sequence.  

However, to use these packages we would need to learn their access functions, data conventions  
(e.g. start/end rules) and GRanges, which would require some additional learning.  To keep things  
as simple as possible, I avoided using these packages (which otherwise would be very relevant for the task).  

# start_section

```{r start_section}

# Start time
Sys.time()

# Initial clenan-up
rm(list=ls())
graphics.off()

# Working folder
base_folder="/home/ec2-user/scripts/data_analysis/ccds_length"
resource_folder="/home/ec2-user/resources/gene_coord"
setwd(base_folder)

# Libraries
library(dplyr)

```

# read_source_data

```{r read_source_data}

# read CCDS data
file_name <- paste(resource_folder,"CCDS.current.txt",sep="/")
CCDS_15_current.df <- read.table(file_name, header=T, sep="\t", comment.char="", as.is = T)
str(CCDS_15_current.df)
dim(CCDS_15_current.df)
CCDS_15_current.df[c(1,6,28,29,101),c(3,10)]
#View(CCDS_15_current.df)

# Read DDR genes
file_name <- paste(resource_folder,"dna_repair_gene_coordinates_grch37_13.txt",sep="/")
ddr_genes.df <- read.table(file_name, header=T, sep=",", as.is = T)
str(ddr_genes.df)
dim(ddr_genes.df)
ddr_genes.df[1:5,]

# Read OR genes
file_name <- paste(resource_folder,"or_gene_coordinates_grch37_13.txt",sep="/")
or_genes.df <- read.table(file_name, header=T, sep=",", as.is = T)
str(or_genes.df)
dim(or_genes.df)
or_genes.df[1:5,]

# Clean-up
rm(file_name)

```

# select_CCDS_for_DDR_genes

```{r select_CCDS_for_DDR_genes}

# Explore genes lists, their intersects and dfferences
DDR_genes <- ddr_genes.df$Gene.name
length(DDR_genes)

CCDS_genes <- unique(CCDS_15_current.df$gene)
length(CCDS_genes)

DDR_CCDS_genes <- intersect(DDR_genes, CCDS_genes)
length(DDR_CCDS_genes)

lost_DDR_genes <- setdiff(DDR_genes, CCDS_genes)
lost_DDR_genes

# Select DDR CCDS data (filter by ssds status and remove not needed fields)
DDR_CCDS_15.df <- CCDS_15_current.df %>% 
  filter(gene %in% DDR_CCDS_genes) %>% 
  filter(ccds_status == "Public") %>% 
  select(-ccds_status, -match_type)

dim(DDR_CCDS_15.df)
str(DDR_CCDS_15.df)
#View(DDR_CCDS_15.df)

# Clean-up
rm(DDR_genes, CCDS_genes, DDR_CCDS_genes, lost_DDR_genes)

```

# select_CCDS_for_OR_genes

Note "ene.name" column ...

```{r select_CCDS_for_OR_genes}

# Explore genes lists, their intersects and dfferences
OR_genes <- or_genes.df$Gene.name # "Gene.name" column ...
length(OR_genes)

CCDS_genes <- unique(CCDS_15_current.df$gene)
length(CCDS_genes)

OR_CCDS_genes <- intersect(OR_genes, CCDS_genes)
length(OR_CCDS_genes)

lost_OR_genes <- setdiff(OR_genes, CCDS_genes)
length(lost_OR_genes)
lost_OR_genes

# Select DDR CCDS data (filter by ssds status and remove not needed fields)
OR_CCDS_15.df <- CCDS_15_current.df %>% 
  filter(gene %in% OR_CCDS_genes) %>% 
  filter(ccds_status == "Public") %>% 
  select(-ccds_status, -match_type)

dim(OR_CCDS_15.df)
str(OR_CCDS_15.df)
#View(OR_CCDS_15.df)

# Clean-up
rm(OR_genes, CCDS_genes, OR_CCDS_genes, lost_OR_genes)

```

# check_missed_DDR_genes

Three DDR genes are missed in CCDS table (PMS2P3, PRKDC, RECQL4).  
These DDR genes should be excluded, when counting coding varaints.  

```{r check_missed_DDR_genes}

# List of initially selected DDR genes
all_ddr_genes <- unique(ddr_genes.df$Gene.name)
length(all_ddr_genes)

# List of DDR genes found in CCDS table
ccds_ddr_genes <- unique(DDR_CCDS_15.df$gene)
length(ccds_ddr_genes)

# Genes missed in CCDS table
setdiff(all_ddr_genes, ccds_ddr_genes)

# Cleanup
rm(all_ddr_genes, ccds_ddr_genes)

```

# check_missed_OR_genes

Nine OR genes are missed in CCDS table:  
OR10G6, OR11H2, OR14A2, OR14K1, OR1F12, OR2AJ1, OR2T7, OR8U9 and OR9G9".    
These OR genes should be excluded, when counting coding varaints.  

```{r check_missed_OR_genes}

# List of initially selected OR genes
all_or_genes <- unique(or_genes.df$Gene.name)
length(all_or_genes)

# List of OR genes found in CCDS table
ccds_or_genes <- unique(OR_CCDS_15.df$gene)
length(ccds_or_genes)

# Genes missed in CCDS table
setdiff(all_or_genes, ccds_or_genes)
length(setdiff(all_or_genes, ccds_or_genes))

# Cleanup
rm(all_or_genes, ccds_or_genes)

```

# function to calculate CDS length from a string given in CDS table  

CCDS table provides information in a format like this:  **[6703348-6703632, 6703977-6704045, 6704353-6704526]**  
This can be used to calculate the length of CDS for each transcript  

```{r calculate_cds_length_function}

calculate_cds_length.udf <- function(cds_string){
  
  # ETL (remove [] and split into vector)
  cds.str <- sub("\\[","",cds_string)
  cds.str <- sub("\\]","",cds.str)
  cds.vct <- unlist(strsplit(cds.str,", "))
  
  # initialise cds length 
  cds_length <- 0 
  
  # For each part of cds
  for(expr in unlist(cds.vct)){
    
    # subtract ens from starts (by parsing and evaluating the string)
    part_length <- eval(parse(text=expr))
    
    # add to the total cds length
    cds_length <- cds_length + part_length
    
  } 
  
  # Return result (negate: because we subtracted ends from starts)
  return(-cds_length)
  
}

# Test calculate_cds_length
str <- DDR_CCDS_15.df[1,"cds_locations"]
calculate_cds_length.udf(str) # 539
calculate_cds_length.udf(DDR_CCDS_15.df[2,"cds_locations"]) # 412
rm(str)

```

# calculate_DDR_cds_length    

```{r calculate_DDR_cds_length}

# Calculate cds length for each transcript in CCDS-15 DDR genes
# Note that sapply function is needed to apply udf to vector in this case  
cds_length <- sapply(DDR_CCDS_15.df$cds_locations,
                     calculate_cds_length.udf)

# Add cds lengths to the table
DDR_CCDS_15.df <- cbind(DDR_CCDS_15.df, cds_length)
str(DDR_CCDS_15.df)

# Select_DDR_transcripts_with_longest_cds
ddr_max_cds_length.df <- DDR_CCDS_15.df %>% 
  arrange(gene) %>% 
  group_by(gene) %>% 
  summarise(max_cds_length=max(cds_length))

# Calculate total length of cds for DDR genes
sum(ddr_max_cds_length.df$max_cds_length)

# Clean-up
rm(cds_length)

```

# calculate_OR_cds_length    

```{r calculate_OR_cds_length}

# Calculate cds length for each transcript in CCDS-15 OR genes
cds_length <- sapply(OR_CCDS_15.df$cds_locations, calculate_cds_length.udf)

# Add cds lengths to the table
OR_CCDS_15.df <- cbind(OR_CCDS_15.df, cds_length)
str(OR_CCDS_15.df)

# Select_OR_transcripts_with_longest_cds
or_max_cds_length.df <- OR_CCDS_15.df %>% 
  arrange(gene) %>% 
  group_by(gene) %>% 
  summarise(max_cds_length=max(cds_length))

# Calculate total length of cds for OR genes
sum(or_max_cds_length.df$max_cds_length)

# Clean-up
rm(cds_length, calculate_cds_length.udf)

```

# save_data

```{r save_data}

save.image(paste(resource_folder, "calculate_CCDS_length.RData", sep="/"))

```

# final_section

```{r final_section}

ls()
sessionInfo()
Sys.time()

```
