---
title: "r01_af_filtering_icr1000_or"
output: html_document
editor_options: 
  chunk_output_type: console
---

Started: David Gilfoyle 3 May 2018  
Updated: David Gilfoyle 16 May 2018

---

# Summary

Variant filtering of GQ filtered dataset based on AF <= 0.01  
Dataset: ICR1000  
Gene Selection: Olfactory Receptor Genes  
Input Variants: 4,200

# start_section

```{r start_section}

Sys.time()
rm(list=ls())
graphics.off()

# Install libraries
library(dplyr)
library(knitr)

```

# load_rdata_file

```{r load_rdata_file}

# Setup input folder
input_folder="/home/ec2-user/results/filtered/gq_filt/icr1000"

# Load RData file - contains dataframe containing all variants
load(paste(input_folder, "r01_icr1000_or_gq_filt.RData", sep="/"))

# Setup script input dataframe naming
input.df <- fixed_crfilt_icr1000_or.df
rm(fixed_crfilt_icr1000_or.df)
input.mx <- gt_add_crfilt_icr1000_or.mx
rm(gt_add_crfilt_icr1000_or.mx)

# Check input dataframe and matrix
dim(input.df)
class(input.df)
input.df[1:20,1:10]
class(input.df$new_AF)
input.df$new_AF[1:20]

dim(input.mx)
class(input.mx)
input.mx[1:5,1:5]

# Check rowmnames are same 
sum(rownames(input.mx) != rownames(input.df)) # should be 0

# Setup base and output folders
base_folder="/home/ec2-user/scripts/filtration/af_filt/icr1000"
output_folder="/home/ec2-user/results/filtered/af_filt/icr1000"

getwd()
setwd(base_folder)
getwd()

opts_knit$set(root.dir = base_folder)

```

# af_filtering

Allele frequency filtering based on selection of rare variants (new_AF <= 0.01) within data set

```{r af_filtering}

# Filter AF <= 0.01
input.df <- tibble::rownames_to_column(input.df, var = "rowname")
output.df <- input.df %>% 
  filter(as.numeric(new_AF) <= 0.01)
output.df <- tibble::column_to_rownames(output.df, var = "rowname")

# Check output dataframe
dim(output.df)
class(output.df)
output.df[1:20,1:10]
output.df$new_AF[1:20]

```

# syncronise_filtered_variants_and_genotype_data 

```{r syncronise_qual_filtered_variants_and_genotype_data}

# Isolate relavent rows from genotype matrix
output.mx <- input.mx[rownames(output.df),]

# Compare filtered fixed column dataframe and genotype matrix
output.df[1:20,1:5]
output.mx[1:20,1:2]

```

# update_dataframe_naming

```{r update_dataframe_naming}

# Reset of output dataframe names
fixed_af_filt_icr1000_or.df <- output.df
gt_add_af_filt_icr1000_or.mx <- output.mx

# Clean-up
rm(input.df, output.df, input.mx, output.mx)

```

# save_results

```{r save_results}

save.image(paste(output_folder, "r01_icr1000_or_af_filtered.RData", sep="/"))

```

# final_section

```{r final_section}

ls()
sessionInfo()
Sys.time()

```

