---
title: "r01_mis_filtering_exac_full_ddr"
output: html_document
editor_options: 
  chunk_output_type: console
---

Started: David Gilfoyle 3 May 2018  
Updated: David Gilfoyle 16 May 2018

---

# Summary

Variant filtering of AF filtered dataset to select missense (Consequence, SIFT [deleterious] and PolyPhen [likely_damaging]), high confidence (HC) missense (missense variants with FATHMM < -1.5 and CADD_PHRED > 20) and HC missense variants in genes with pLI >= 0.9  
Dataset: ExAC (full)  
Gene Selection: DNA Repair Genes  
Input Variants: 99,236

# start_section

```{r start_section}

Sys.time()
rm(list=ls())
graphics.off()

# Install libraries
library(dplyr)
library(knitr)

```

# load_rdata_file

```{r load_rdata_file}

# Setup input folder
input_folder="/home/ec2-user/results/filtered/af_filt/exac_v1"

# Load RData file - contains dataframe containing all variants
load(paste(input_folder, "r01_exac_full_ddr_af_filtered.RData", sep="/"))

# Setup script input dataframe naming
input.df <- fixed_af_filt_exac_full_ddr.df
rm(fixed_af_filt_exac_full_ddr.df)

# Check input dataframe
dim(input.df)
class(input.df)
input.df[1:20,1:10]
class(input.df$Consequence)
input.df$Consequence[1:20]
class(input.df$FATHMM)
input.df$FATHMM[500:570]
class(input.df$CADD_PHRED)
input.df$CADD_PHRED[1:20]
class(input.df$ExACpLI)
input.df$ExACpLI[1:20]

# Setup base and output folders
base_folder="/home/ec2-user/scripts/filtration/mis_filt/exac_v1"
output_folder="/home/ec2-user/results/filtered/mis_filt/exac_v1"

getwd()
setwd(base_folder)
getwd()

opts_knit$set(root.dir = base_folder)

```

# variants_associated_with_ddr_genes

```{r variants_associated_with_ddr_genes}

# Set of DDR genes supporting variant filtration
ddr_genes <- c("UNG", "SMUG1", "MBD4", "TDG", "OGG1", "MUTYH", "NTHL1", "MPG", "NEIL1", "NEIL2", "NEIL3", "HMGB1", "APEX1", "APEX2", "LIG3", "XRCC1", "PNKP", "APLF", "PARP1", "PARP2", "PARP3", "MGMT", "ALKBH2", "ALKBH3", "TDP1", "TDP2", "MSH2", "MSH3", "MSH6", "MLH1", "PMS2", "MSH4", "MSH5", "MLH3", "PMS1", "PMS2P3", "XPC", "RAD23B", "CETN2", "RAD23A", "XPA", "DDB1", "DDB2", "RPA1", "RPA2", "RPA3", "ERCC3", "ERCC2", "GTF2H1", "GTF2H2", "GTF2H3", "GTF2H4", "GTF2H5", "CDK7", "CCNH", "MNAT1", "ERCC5", "ERCC1", "ERCC4", "LIG1", "RBX1", "CUL4A", "CUL4B", "ERCC8", "ERCC6", "UVSSA", "XAB2", "MMS19", "RAD51", "RAD51B", "RAD51D", "DMC1", "XRCC2", "XRCC3", "RAD52", "RAD54L", "RAD54B", "BRCA1", "SHFM1", "RAD50", "MRE11A", "NBN", "RBBP8", "MUS81", "EME1", "EME2", "SLX1A", "SLX1B", "GEN1", "BARD1", "BABAM1", "FAM175A", "BRE", "UIMC1", "BRCC3", "SYCP3", "FANCA", "FANCB", "FANCC", "BRCA2", "FANCD2", "FANCE", "FANCF", "FANCG", "FANCI", "BRIP1", "FANCL", "FANCM", "PALB2", "RAD51C", "SLX4", "C1orf86", "C19orf40", "APITD1", "WDR48", "USP1", "C17orf70", "HES1", "RMI1", "RMI2", "TOP3A", "UBE2T", "XRCC6", "XRCC5", "PRKDC", "LIG4", "XRCC4", "DCLRE1C", "NHEJ1", "DNTT", "NUDT1", "DUT", "RRM2B", "POLB", "POLG", "POLD1", "POLE", "PCNA", "REV3L", "MAD2L2", "REV1", "POLH", "POLI", "POLQ", "POLK", "POLL", "POLM", "POLN", "FEN1", "FAN1", "TREX1", "TREX2", "EXO1", "APTX", "SPO11", "ENDOV", "UBE2A", "UBE2B", "RAD18", "SHPRH", "HLTF", "RNF168", "SPRTN", "RNF8", "RNF4", "UBE2V2", "UBE2N", "H2AFX", "CHAF1A", "SETMAR", "BLM", "WRN", "RECQL4", "ATM", "MPLKIP", "DCLRE1A", "DCLRE1B", "RPA4", "PRPF19", "RECQL", "RECQL5", "HELQ", "RDM1", "NABP2", "RFC1", "RFC2", "RFC3", "RFC4", "RFC5", "ATR", "ATRIP", "MDC1", "RAD1", "RAD9A", "HUS1", "RAD17", "CHEK1", "CHEK2", "TP53", "TP53BP1", "RIF1", "TOPBP1", "CLK2", "PER1")

# Filtering of variants based on DDR genes using SYMBOL 
output.df <- input.df %>%
  filter(input.df$SYMBOL %in% ddr_genes)

removed_variant.df <- input.df %>%
  filter(!input.df$SYMBOL %in% ddr_genes)

# Check output dataframe
dim(output.df)
class(output.df)
output.df[1:20,1:10]
class(input.df$SYMBOL)
unique(output.df$SYMBOL)
unique(removed_variant.df$SYMBOL)

```

# mis_filtering

Filtering based on VEP consequences, SIFT and PolyPhen scores:  
1) Consequence - missense_variant  
2) SIFT - deleterious (removing deleterious_low_confidence)  
3) PolyPhen - probably_damaging

```{r mis_filtering}

# Assessment of SIFT and PolyPhen
sum(is.na(input.df$SIFT))
sum(is.na(input.df$PolyPhen))

# Filtration based on consequence, SIFT and PolyPhen scores
output.df <- output.df %>% 
  filter(grepl("missense_variant",Consequence)) %>%
  filter(grepl("deleterious", SIFT)) %>%
  filter(!grepl("deleterious_low_confidence", SIFT)) %>%
  filter(grepl("probably_damaging", PolyPhen))

# Check output dataframe
dim(output.df)
output.df[1:20,1:10]
output.df$Consequence[1:20]
output.df$SIFT[1:20]
output.df$PolyPhen[1:20]
output.df$FATHMM[1:20]
output.df$CADD_PHRED[1:20]

```

# additional_filtering_fathmm_cadd

Output assessed based on further filtering with:  
1) FATHMM < -1.5 - FATHMM contains score and information - needs to be split prior to filtering. A relatively low number of FATHMM entries are "Prediction(Available)" - status of these variants is unknown, but there is no obvious reason for the lack of score. However, in order to enrich the dataset for deleterious missense variants, these variants have been removed (appear as NA in seperated FATHMM_SCORE column)  
2) CADD_PHRED > 20 - assessment indicates that some entries are NA - these appear to be indels (substitution variants) and will be retained. However, removal of variants with CADD_PHRED <= 20 is considered appropriate to enrich dataset for potentially pathogenic variants  

```{r additional_filtering_fathmm_cadd}

# Assessment of FATHMM column in output dataframe
sum(is.na(output.df$FATHMM))
sum(output.df$FATHMM == "Prediction(Available)")
sum(output.df$FATHMM == -1.5)
sum(output.df$FATHMM > -1.5)
sum(output.df$FATHMM < -1.5)

# output_fathmm_pa.df <- output.df %>% 
#  filter(FATHMM == "Prediction(Available)")

# Seperate FATHMM score and info into two seperare columns
output_cor.df <-
  output.df %>% 
  mutate(FATHMM_SCORE=as.numeric(
    sub("\\(.*\\)","",FATHMM))) %>% 
  mutate(FATHMM_INFO=
    sub(".*\\(","", sub("\\)","",FATHMM))) %>% 
  select(-FATHMM)

# Assessment of FATHMM columns and impact of filtering FATHMM_SCORE < -1.5 in seperated dataframe
class(output_cor.df$FATHMM_SCORE)
class(output_cor.df$FATHMM_INFO)
sum(is.na(output_cor.df$FATHMM_SCORE))
sum(output_cor.df$FATHMM_SCORE == -1.5, na.rm = TRUE)
sum(output_cor.df$FATHMM_SCORE > -1.5, na.rm = TRUE)
sum(output_cor.df$FATHMM_SCORE < -1.5, na.rm = TRUE)

# Assessment of impact of filtering CADD_PHREDD > 20 in seperated dataframe
sum(is.na(output_cor.df$CADD_PHRED))
sum(output_cor.df$CADD_PHRED == 20, na.rm = TRUE)
sum(output_cor.df$CADD_PHRED < 20, na.rm = TRUE)
sum(output_cor.df$CADD_PHRED > 20, na.rm = TRUE)

# output_cadd_na.df <- output.df %>% 
#  filter(is.na(CADD_PHRED))

# Additional filtration based on FATHMM_SCORE and CADD_PHRED
output_hc.df <- output_cor.df %>% 
  filter(FATHMM_SCORE < -1.5) %>%
  filter((as.numeric(CADD_PHRED) > 20) |
           (is.na(CADD_PHRED)))

# Check output dataframe
dim(output_hc.df)
output_hc.df[1:20,1:10]
output_hc.df$Consequence[1:20]
output_hc.df$FATHMM_SCORE[1:20]
output_hc.df$CADD_PHRED[100:120] # range updated to show presence of NA
unique(output_hc.df$SYMBOL)

```

# additional_filtering_exac_pli

Confirm HC missense variants in loss intolerant genes (ExAC pLI >= 0.9) 

```{r additional_filtering_exac_pli}

# Assessment of impact of filtering ExAC pLI >= 0.9
sum(is.na(output_hc.df$ExACpLI))
sum(as.numeric(output_hc.df$ExACpLI) < 0.9, na.rm = TRUE)
sum(as.numeric(output_hc.df$ExACpLI) >= 0.9, na.rm = TRUE)

# Additional filtration based on ExAC pLI >= 0.9
output_hc_pli.df <- output_hc.df %>% 
  filter(as.numeric(ExACpLI) >= 0.9)

# Check output dataframe
dim(output_hc_pli.df)
output_hc_pli.df[1:20,1:10]
output_hc_pli.df$Consequence[1:20]
output_hc_pli.df$FATHMM_SCORE[1:20]
output_hc_pli.df$CADD_PHRED[1:20]
output_hc_pli.df$ExACpLI[1:20]
unique(output_hc_pli.df$SYMBOL)

```

# update_dataframe_naming

```{r update_dataframe_naming}

# Reset of output dataframe names
fixed_mis_filt_exac_full_ddr.df <- output.df
fixed_mis_hc_filt_exac_full_ddr.df <- output_hc.df
fixed_mis_hc_pli_filt_exac_full_ddr.df <- output_hc_pli.df

# Clean-up
rm(input.df, output.df, output_hc.df, output_hc_pli.df, ddr_genes, removed_variant.df)

```

# save_results

```{r save_results}

save.image(paste(output_folder, "r01_exac_full_ddr_mis_filtered.RData", sep="/"))

```

# final_section

```{r final_section}

ls()
sessionInfo()
Sys.time()

```

